Fix (initial) melodic-devel branch

henningkayser opened the pull request:
Applies necessary changes to new melodic branch.

* update references melodic->kinetic
* refactor Affine3d -> Isometry3d

henningkayser commented:
Travis CI fails because this PR depends on https://github.com/PickNikRobotics/rosparam_shortcuts/pull/7 which is not merged yet.

mlautman commented:
@henningkayser I merged the rosparam_shortcuts pr but this still fails. I am sure that it is because a new debian would need to be released. One way to show this would be to add a moveit_grasps.rosinstall and configure moveit_ci to use the new rosinstall so that rosparam_shortcuts gets built from source. I think that it is important to get CI passing before we can merge this

henningkayser commented:
> @henningkayser I merged the rosparam_shortcuts pr but this still fails. I am sure that it is because a new debian would need to be released. One way to show this would be to add a moveit_grasps.rosinstall and configure moveit_ci to use the new rosinstall so that rosparam_shortcuts gets built from source. I think that it is important to get CI passing before we can merge this

@mlautman Looks like you are right. However, I wouldn't add a .rosinstall since we want to release `moveit_grasps` as well. When can we make the next `rosparam_shortcuts` release? I could also load the parameters as Affine3d and convert them to Isometry3d.

mlautman commented:
@henning I'm not convinced that we need to test against both the outdated debians and the newest debians. I'm pretty sure we can safely remove line 18 `- ROS_DISTRO=melodic ROS_REPO=ros` from the `.travis.yaml` file. This will fix ci.

mlautman commented on .travis.yml:
```diff
@@ -14,9 +14,9 @@ env:
-    - ROS_DISTRO=kinetic ROS_REPO=ros
-    - ROS_DISTRO=kinetic ROS_REPO=ros-shadow-fixed
+    - ROS_DISTRO=melodic ROS_REPO=ros              TEST=clang-format
+    - ROS_DISTRO=melodic ROS_REPO=ros
```
I'm not sure why we are testing agianst both the `ci` and `shadow-fixed-ci` debians but I am fairly confident that we can remove this failing test.

gavanderhoorn replied:
> I'm not sure why we are testing agianst both the `ci` and `shadow-fixed-ci` debians

it's a form of (almost reverse) regression testing: new changes in other packages (staged in the shadow repository) will be tested against the current state of your package(s).

Without that test you'll only know something is broken (either in your package(s) or in others) *after* everything has been synced to the main repositories.

Probably not what you want.

mlautman replied:
Another possible fix would be to change the rosparam_shortcuts build_depend to  `<build_depend version_gte="0.3.1">rosparam_shortcuts</build_depend>`

gavanderhoorn replied:
I'm not sure how that would fix things. If the package version that this depends on is only available in the staging repository, nothing will change (`version` specs in manifests are not checked by CMake/Catkin).

mlautman replied:
I guess that would be a regression. 

I was hoping that we could use the rosdep install call in moveit_ci to update rosparam_shortcuts to 0.3.1 but that does not seem to work on my local machine.

mlautman commented on README.md:
```diff
@@ -26,30 +26,30 @@ Developed by Dave Coleman, Andy McEvoy, and Mike Lautman at [PickNik Consulting]
 ## Robot-Agnostic Configuration
 
-You will first need a configuration file that described your robot's end effector geometry. Currently an example format can be seen in this repository at [config_robot/baxter_grasp_data.yaml](https://github.com/PickNikRobotics/moveit_grasps/blob/kinetic-devel/config_robot/baxter_grasp_data.yaml). See the comments within that file for explanations.
+You will first need a configuration file that described your robot's end effector geometry. Currently an example format can be seen in this repository at [config_robot/baxter_grasp_data.yaml](https://github.com/ros-planning/moveit_grasps/blob/melodic-devel/config_robot/baxter_grasp_data.yaml). See the comments within that file for explanations.
```
My bad but this should be switched to `panda_grasp_data.yaml` in both the melodic and kinetic branches

henningkayser replied:
I think we should remove these instructions and only link the tutorial https://github.com/ros-planning/moveit_tutorials/pull/302.

mlautman commented on README.md:
```diff
@@ -107,11 +107,11 @@ When filtered, the colors represent the following:
 There are other example implementations:
 
- - [baxter_pick_place](https://github.com/davetcoleman/baxter_cpp/tree/kinetic-devel/baxter_pick_place)
+ - [baxter_pick_place](https://github.com/davetcoleman/baxter_cpp/tree/melodic-devel/baxter_pick_place)
```
broken link. Please revert. Also, this isn't built in that repo so I'm hesitant to include it in this repo. Maybe just remove

henningkayser replied:
same. I could replace config and demo instructions by a link to the not-yet merged tutorial

mlautman commented on README.md:
```diff
@@ -122,7 +122,7 @@ To run [roslint](http://wiki.ros.org/roslint), use the following command with [c
 To run [catkin lint](https://pypi.python.org/pypi/catkin_lint), use the following command with [catkin-tools](https://catkin-tools.readthedocs.org/).
 
-    catkin lint -W2 --rosdistro kinetic
+    catkin lint -W2 --rosdistro melodic
```
Nit: If you change all of these commands from `melodic` to `$ROS_DISTRO` it will reduce maintenance

henning commented:
No idea how i can help you guys... wrong username?


On Mon 15. Apr 2019 at 21:34, Mike Lautman <notifications@github.com> wrote:

> @henning <https://github.com/henning> I'm not convinced that we need to
> test against both the outdated debians and the newest debians. I'm pretty
> sure we can safely remove line 18 - ROS_DISTRO=melodic ROS_REPO=ros from
> the .travis.yaml file. This will fix ci.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ros-planning/moveit_grasps/pull/35#issuecomment-483387651>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AACdUDRgnjJorhY5aw7FMujt_yzr70K5ks5vhNQ0gaJpZM4ckWc0>
> .
>
-- 
Henning Sprang
http://www.sprang.de

mlautman commented:
@henning my bad, I meant to tag @henningkayser. Kinda crazy that github doesn't order suggested mentions based on actual contributors lol

henningkayser commented:
@mlautman I fixed CI by adding a .rosinstall file and updated the README.
I will add a new PR for cleaning up the README and link the tutorial so we don't have duplicate instructions.
